packages/gatsby/src/schema/types/NodeInterface.js

```diff
@@ -0,0 +1,76 @@
+const getOrCreateNodeInterface = schemaComposer => {
+  // TODO: why is `mediaType` on Internal? Applies only to File!?
+  // Is `fieldOwners` actually set anywhere? And is it supposed to be an array,
+  // as the Joi schema claims, or an object, which is the default in `redux/actions.js`
+  const InternalTC = schemaComposer.getOrCreateTC(`Internal`, tc => {
+    tc.addFields({
+      content: `String`,
+      contentDigest: `String`,
+      description: `String`,
+      fieldOwners: [`String`],
+      ignoreType: `Boolean`,
+      mediaType: `String`,
+      owner: `String`,
+      type: `String!`,
+    })
+  })
+  InternalTC.getITC()
```

Comment:
Probably not needed now that we use toInputObjectType directly (same with `NodeInterface.getITC()`) - We called this here to register a non-prefixed name (`Nodenput` and not `MarkdownParentNodeInput` - but we should try to remove prefixing upstream)

Reply:
What do you mean by "remove prefixing upstream"?

Reply:
Just not adding the type name [here](https://github.com/graphql-compose/graphql-compose/blob/364dcdd2c483ba81d99dfd6c784823f6555eaea4/src/utils/toInputObjectType.js#L105)

Reply:
One thing I was thinking about is introducing concept of a "TypeSetComposer" which is an abstraction on top of TypeComporers, InputTypeComposers etc and allows defining dependencies and transformations. Eg we can then write own getITC that knows how to not prefix. This would also solve the issue with clearing the ITC cache.

Reply:
I'll have to think a bit about this, but not sure if we need to do much more than we already have: can we just call `tc.removeInputTypeComposer(); tc.getITC()` [here](https://github.com/gatsbyjs/gatsby/blob/9a59b474bb2e663f4a16f1ca53f469c6cf41c32b/packages/gatsby/src/schema/schema.js#L246)? What other issues would TypeSetComposer address?

Reply:
I thought it might also have all the other type composers linked, like filter, sort etc. So we could clear all the generated types in one go when original type changes. Could be useful for things like live schema reload.